### PR TITLE
Contract/examples

### PR DESCRIPTION
**Pull request by rpfister102**

### CODE REVIEW

File: examples/Ballot.bl [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    [Pre=voters[voter].weight == 0]
+    function void giveRightToVote(address voter) {
+        voters[voter].weight = 1
+    }
+
+    [Pre = !voters[msg.sender].voted]
+    [Pre = to != msg.sender]
+    function void delegate(address to) {
+        Voter sender = voters[msg.sender]
+        to_ = to
```
Comment (tk-codes) [inline examples/Ballot.bl]: `to` and `to_` can lead to careless mistakes.
I would rename
* `to` --> `initialTo`
* `to_` --> `currentTo`

File: examples/Ballot.bl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        chairperson = msg.sender
+        voters[chairperson].weight = 1
+        proposals = new Proposal[proposalNames.length]
+
+        foreach(i, String name : proposalNames) {
+            proposals[i] = new Proposal(name, 0)
+        }
+    }
+
+    [Pre=msg.sender == chairperson]
```
Comment (tk-codes) [inline examples/Ballot.bl]: Use built-in Annotation: `[Owner]`

File: examples/Ballot.bl [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    [Pre=!voters[voter].voted]
```
Comment (tk-codes) [inline examples/Ballot.bl]: Code style: Use spaces near `=`

File: examples/Ballot.bl [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                break
+            }
+            to_ = voters[to_].delegate
+
+            assert(to_ != msg.sender) // Found loop in delegation
+        }
+
+        sender.voted = true
+        sender.delegate = to_
+        Voter delegate_ = voters[to_]
```
Comment (tk-codes) [inline examples/Ballot.bl]: Rename `delegate_` to `delegatedVoter`

File: examples/Ballot.bl [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        sender.voted = true
+        sender.vote = proposal
+
+        proposals[proposal].voteCount += sender.weight
+    }
+
+    readonly function int winningPoposal() {
+            int winningVoteCount = 0
+            int winningProposal = 0
+            for (int p : 0 to proposals.length - 1) {
```
Comment (tk-codes) [inline examples/Ballot.bl]: Use `foreach` loop

File: examples/Ballot.bl [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                if (proposals[p].voteCount > winningVoteCount) {
+                    winningVoteCount = proposals[p].voteCount
+                    winningPoposal = p
+                }
+            }
+            return winningPoposal
+    }
+
+    readonly function String winnerName() {
```
Comment (tk-codes) [inline examples/Ballot.bl]: readonly is annotation. so `[ReadOnly]`

File: examples/Ballot.bl [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    struct Voter {
+        int weight
+        bool voted
+        address delegate
+        int vote
+    }
+
+    struct Proposal {
+        String name
```
Comment (tbocek) [inline examples/Ballot.bl]: minor comment, these primitives: int, bool, address are lowercase. But String is uppercase.

File: examples/Ballot.bl [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    address chairperson
+
+    Map<address, Voter> voters
+
+    Proposal[] proposals
+
+    constructor(String[] proposalNames) {
+        chairperson = msg.sender
+        voters[chairperson].weight = 1
+        proposals = new Proposal[proposalNames.length]
```
Comment (tbocek) [inline examples/Ballot.bl]: do we need a "new" keyword and do we support dynamic length?

File: examples/Ballot.bl [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    [Pre=msg.sender == chairperson]
+    [Pre=!voters[voter].voted]
+    [Pre=voters[voter].weight == 0]
+    function void giveRightToVote(address voter) {
+        voters[voter].weight = 1
+    }
+
+    [Pre = !voters[msg.sender].voted]
```
Comment (tbocek) [inline examples/Ballot.bl]: This colud look confusing, one might want to write Pre != voters... What about Pre:

File: examples/Ballot.bl (hunk 1 above)
Comment (tbocek) [reply examples/Ballot.bl]: no type required for to_ or currentTo? I'm fine with that, as we know to is an address

File: examples/BlindAuction.bl [hunk 10]
```diff
@@ -0,0 +1,102 @@
+version 1.1
+
+contract BlindAuction {
+
+    struct Bid {
+        String blindedBid
+        int deposit
+    }
+
+    payableaddress beneficiary
```
Comment (tbocek) [inline examples/BlindAuction.bl]: new type?

File: examples/BlindAuction.bl [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    Map<address, Bid[]> bids
+
+    address highestBidder
+    int highestBid
+
+    Map<address, int> pendingReturns
+
+    event AuctionEnded(address winner, int highestBid)
+
+    function bool onlyBefore(int time) {
```
Comment (tbocek) [inline examples/BlindAuction.bl]: this can be internal, right?

File: examples/Ballot.bl (hunk 1 above)
Comment (tk-codes) [reply examples/Ballot.bl]: We don't support type inference in the first version. Therefore type is also required here.

File: examples/Ballot.bl (hunk 7 above)
Comment (tk-codes) [reply examples/Ballot.bl]: Value types are lowercase and reference types are uppercase. That's why String is uppercase here.

File: examples/Ballot.bl (hunk 8 above)
Comment (tk-codes) [reply examples/Ballot.bl]: `new` keyword is required to differentiate between index access and array creation in Parser.
We don't support dynamic length. Is `slice/list` required?

**Pull request merged**